src/neoxp/Commands/ContractCommand.Download.cs

```diff
@@ -0,0 +1,76 @@
+using System;
+using System.ComponentModel.DataAnnotations;
+using System.IO;
+using System.Net.Http;
+using System.Threading.Tasks;
+using McMaster.Extensions.CommandLineUtils;
+using Neo;
+using Neo.IO.Json;
+using Neo.Network.RPC;
+
+namespace NeoExpress.Commands
+{
+    partial class ContractCommand
+    {
+        [Command(Name = "download", Description = "Download contract with storage from remote chain into local chain")]
+        internal class Download
+        {
+            readonly ExpressChainManagerFactory chainManagerFactory;
+            
+            public Download(ExpressChainManagerFactory chainManagerFactory)
+            {
+                this.chainManagerFactory = chainManagerFactory;
+            }
+            
+            [Argument(0, Description = "Contract invocation hash")]
+            [Required]
+            internal string Contract { get; init; } = string.Empty;
+            
+            [Argument(1, Description = "URL of Neo JSON-RPC Node\nSpecify MainNet (default), TestNet or JSON-RPC URL")]
+            internal string RpcUri { get; } = string.Empty;
+            
+            [Option(Description = "Path to neo-express data file")]
+            internal string Input { get; init; } = string.Empty;
+
+            internal async Task ExecuteAsync(TextWriter writer)
+            {
+                var (chainManager, _) = chainManagerFactory.LoadChain(Input);
+                var expressNode = chainManager.GetExpressNode();
+
+                if (!UInt160.TryParse(Contract, out var contractHash))
+                    throw new ArgumentException($"Invalid contract hash: \"{Contract}\"");
+
+                if (!TransactionExecutor.TryParseRpcUri(RpcUri, out var uri))
+                    throw new ArgumentException($"Invalid RpcUri value \"{RpcUri}\"");
```

Comment:
Let me know if I should add `{}` around the throw. I noticed a mixed style for single line statements after a condition (with and without braces). This style came from one of the policy commands I copied the RpcUri parsing from.

Reply:
THanks for asking. I typically only skip braces when the entire if/then statement "looks good" on a single line. I'd appreciate if you'd add braces for these